src/cclib/parser/orcaparser.py

```diff
@@ -643,23 +643,144 @@ def extract(self, inputfile, line):
                 self.etsecs.append(sec)
                 line = next(inputfile)
 
-        # This will parse etoscs for TD calculations, but note that ORCA generally
-        # prints two sets, one based on the length form of transition dipole moments,
-        # the other based on the velocity form. Although these should be identical
-        # in the basis set limit, in practice they are rarely the same. Here we will
-        # effectively parse just the spectrum based on the length-form.
-        if (line[25:44] == "ABSORPTION SPECTRUM" or line[9:28] == "ABSORPTION SPECTRUM") and not hasattr(self, "etoscs"):
-
-            self.skip_lines(inputfile, ['d', 'header', 'header', 'd'])
-
-            self.etoscs = []
-            for x in self.etsyms:
-                osc = next(inputfile).split()[3]
-                if osc == "spin":  # "spin forbidden"
-                    osc = 0
-                else:
-                    osc = float(osc)
-                self.etoscs.append(osc)
+
+        # Parse the various absorption spectra for TDDFT and ROCIS
+        if 'ABSORPTION SPECTRUM' in line or 'ELECTRIC DIPOLE' in line:
+
+            # Standard header, occasionally changes
+            header = ['d', 'header', 'header', 'd']
+
+            def energy_intensity(line):
+                """ TDDFT and related methods standard method of output
+-----------------------------------------------------------------------------
+         ABSORPTION SPECTRUM VIA TRANSITION ELECTRIC DIPOLE MOMENTS
+-----------------------------------------------------------------------------
+State   Energy  Wavelength   fosc         T2         TX        TY        TZ
+        (cm-1)    (nm)                  (au**2)     (au)      (au)      (au)
+-----------------------------------------------------------------------------
+   1 5184116.7      1.9   0.040578220   0.00258  -0.05076  -0.00000  -0.00000
+"""
+                try:
+                    state, energy, wavelength, intensity, t2, tx, ty, tz = line.split()
+                except ValueError as e:
+                    # Must be spin forbidden and thus no intensity
+                    energy = line.split()[1]
+                    intensity = 0
+                return energy, intensity
+
+            # Check for variations
+            if (line[16:24] == 'COMBINED' and not 'ROCIS' in line) or \
```

Comment:
Is there a reason why this isn't the full string, say `if line.strip() == 'COMBINED ELECTRIC DIPOLE + MAGNETIC DIPOLE + ELECTRIC QUADRUPOLE SPECTRUM'`? Same for the others. If not, this is harder to see the logic for.